src/main/java/org/threadly/litesockets/NoThreadSocketExecuter.java

```diff
@@ -137,69 +153,69 @@ public void select() {
   public void select(final int delay) {
     ArgumentVerifier.assertNotNegative(delay, "delay");
     checkRunning();
-    try {
-      commonSelector.selectNow();  //We have to do this before we tick for windows
-      localNoThreadScheduler.tick(null);
-      if(delay == 0) {
-        commonSelector.selectNow();
-      } else {
-        commonSelector.select(delay);
-      }
-      if(isRunning()) {
-        for(final SelectionKey key: commonSelector.selectedKeys()) {
-          try {
-            if(key.isAcceptable()) {
-              doServerAccept(servers.get(key.channel()));
-            } else {
-              final Client tmpClient = clients.get(key.channel());
-              if(key.isConnectable() && tmpClient != null) {
-                doClientConnect(tmpClient, commonSelector);
-                key.cancel(); //Stupid windows bug here.
-                setClientOperations(tmpClient);
+    long startTime = Clock.accurateForwardProgressingMillis();
+    while(Clock.accurateForwardProgressingMillis()- startTime <= delay && isRunning() && !wakeUp) {
+      try {
+        commonSelector.selectNow();  //We have to do this before we tick for windows
+        localNoThreadScheduler.tick(null);
+        commonSelector.select(Math.min(delay, 50));
```

Comment:
so once select is called the thread is blocked till either the time expires or a socket operation happens.  Since this thread is also running the NoThreadScheduler it could cause things scheduled (watchDog mainly) to get stuck behind it.